### DISCUSSION TITLE
What is the most convenient way to get kitten on the root user  via SSH?

### DISCUSSION THREAD

**Discussion by ecerulm** [kovidgoyal/kitty#7668] [Q&A] [answered]
Currently I can ssh to a machine with 
```
kitten ssh myuser@xxx.example.com
```

and I have `kitten` automatically in the remote shell, but I usually switch to the root user with   `sudo su -` and then both kitten and the terminfo are not available to the `root` user. 

The `TERM` / `TERMINFO` can be preserved as explained in [I get errors about the terminal being unknown or opening the terminal failing or functional keys like arrow keys don’t work?](https://sw.kovidgoyal.net/kitty/faq/#i-get-errors-about-the-terminal-being-unknown-or-opening-the-terminal-failing-or-functional-keys-like-arrow-keys-don-t-work). 

so by adding using `sudo visudo` and adding
```
Defaults env_keep += "TERM TERMINFO"
```

and using `sudo su` instead of `sudo su -` you can have an usable terminal for the root user. But the `kitten` command is not accessible for the root user (`bash: kitten: command not found`). 

What would be the most effective way to replicate what the `kitten ssh` did for the `myuser` but for `root` instead (I can't ssh as root directly).

**Comment by ecerulm** [comment]
I currently do 
* configure sudo to preserve TERM and TERMINFO 
* copy the /home/xxxx/.local/share/kitty-ssh-kitten/ to /root/.local/share/kitty-ssh-kitten/
* edit `~/.bashrc` to set the `$PATH` to include `$HOME/.local/share/kitty-ssh-kitten/bin`


```
sudo visudo # add Defaults env_keep += "TERM TERMINFO"
sudo su # do not use -l
mkdir -p ~/.local/share
cp  -r /home/ec2-user/.local/share/kitty-ssh-kitten/ ~/.local/share/
vim ~/.bashrc 
# add export PATH="$HOME/.local/share/kitty-ssh-kitten/kitty/bin:$PATH"
```

**Comment by kovidgoyal** [comment]
Switch to a root shell with

sudo kitten run-shell

**Reply by ecerulm** [reply]
that is way simpler and it works like a charm!

I had  to use `sudo $(which kitten) run-shell` since the `sudo` uses `secure_path` and `kitten` is not there.

**Reply by ecerulm** [reply]
I realized that this does not have the "kitten" in the path (it did work for me because I already added `/root/.local/share/kitty-ssh-kitten` and added to `$PATH`) 

I tested without that and 

```
sudo $(which kitten) run-shell
kitten @ set-tab-title xxxx # bash: kitten: command not found 
```

**Reply by kovidgoyal** [reply]
https://github.com/kovidgoyal/kitty/commit/bb3aecd2a5fb2b09d71d32432f18cd872da40c09

**Comment by kovidgoyal** [comment, marked as answer]
sudo $(which kitten) run-shell --env=$PATH:$(dirname `which kitten`)

This will bypass sudo's secure_path though, which personally I find
pretty silly, but YMMV.

**Reply by ecerulm** [reply]
I had to modify the `--env` part :  
```
sudo $(which kitten) run-shell --env PATH=$PATH:$(dirname `which kitten`)
```
so instead of `--env=xxx` it has to be `--env=PATH=xxx` or `--env PATH=xxx`
